### PR TITLE
adds functions for getting pk's from sk's

### PR DESCRIPTION
**Pull request by tacitus-aedifex**
@stef this adds the crypto_scalarmult_base function for getting a pk from a crypto_box sk, crypto_sign_ed25519_sk_to_seed and crypto_sign_ed25519_sk_to_pk functions for getting the pk and seed from a crypto_sign sk.  i also added tests to verify.

i needed these functions too.

### CODE REVIEW

File: pysodium/__init__.py [hunk 1]
```diff
@@ -293,6 +293,14 @@ def crypto_box_seed_keypair(seed):
     __check(sodium.crypto_box_seed_keypair(pk, sk, seed))
     return pk.raw, sk.raw
 
+# int crypto_scalarmult_base(unsigned char *q, const unsigned char *n)
+def crypto_scalarmult_base(sk):
```
Comment (stef) [inline pysodium/__init__.py]: what's the difference to crypto_scalarmult_curve25519_base in line 148?

File: pysodium/__init__.py [hunk 2]
```diff
@@ -409,6 +417,15 @@ def crypto_sign_seed_keypair(seed):
     __check(sodium.crypto_sign_seed_keypair(pk, sk, seed))
     return pk.raw, sk.raw
 
+def crypto_sign_ed25519_sk_to_seed(sk):
+    seed = ctypes.create_string_buffer(crypto_sign_SEEDBYTES)
+    __check(sodium.crypto_sign_ed25519_sk_to_seed(seed, sk))
+    return seed.raw
+
+def crypto_sign_ed25519_sk_to_pk(sk):
```
Comment (stef) [inline pysodium/__init__.py]: how is this different from crypto_sign_sk_to_pk in line 522 of the unpatched file?

**Comment by stef** [comment]
seems the only function we didn't have yet is crypto_sign_ed25519_sk_to_seed if i see things correctly. or do i miss something?

**Comment by tacitus-aedifex** [comment]
Huh, interesting.  Did libsodium rename those functions?

**Comment by tacitus-aedifex** [comment]
oops, sorry, i see what's going on now.  let me revise and squash this patch so that it only adds the sk_to_seed.  i'll name it crypto_sign_sk_to_seed to follow the patter of crypto_sign_sk_to_pk.  i'll fix up the tests too.

**Comment by tacitus-aedifex** [comment]
@stef alright, sorry for the confusion.  i think i cleaned it up correctly.  let me know if i need to fix anything.

**Comment by tacitus-aedifex** [comment]
@stef, ping.

**Comment by stef** [comment]
thx!

**Pull request merged**